src/DeviceInterface/ExtensionInterface.js

```diff
@@ -45,39 +47,16 @@
             return TOOLTIP_TYPES.EmailProtection
         }
 
-        if (this.settings.featureToggles.emailProtection_incontext_signup && this.settings.incontextSignupPermanentlyDismissed === false) {
+        if (this.settings.featureToggles.emailProtection_incontext_signup && this.inContextSignup?.permanentlyDismissed === false) {
             return TOOLTIP_TYPES.EmailSignup
         }
 
         return null
     }
 
-    onIncontextSignup () {
-        this.firePixel({pixelName: 'incontext_get_email_protection'})
-    }
-
-    onIncontextSignupDismissed () {
-        // Check if the email signup tooltip has previously been dismissed.
-        // If it has, make the dismissal persist and remove it from the page.
-        // If it hasn't, set a flag for next time and just hide the tooltip.
-        if (this.settings.incontextSignupInitiallyDismissed) {
-            this.settings.setIncontextSignupPermanentlyDismissed(true)
-            this.deviceApi.notify(new SetIncontextSignupPermanentlyDismissedAtCall({ value: new Date().getTime() }))
-            this.removeAutofillUIFromPage()
-            this.firePixel({pixelName: 'incontext_dismiss_persisted'})
-        } else {
-            this.settings.setIncontextSignupInitiallyDismissed(true)
-            this.deviceApi.notify(new SetIncontextSignupInitiallyDismissedAtCall({ value: new Date().getTime() }))
-            this.removeTooltip()
-            this.firePixel({pixelName: 'incontext_dismiss_initial'})
-        }
-    }
-
     async resetAutofillUI (callback) {
         this.removeAutofillUIFromPage()
 
-        // Start the setup process again
-        await this.refreshSettings()
```

Comment:
no need to refresh global settings - anything this 'feature' needs can be done in 'setupAutofill'